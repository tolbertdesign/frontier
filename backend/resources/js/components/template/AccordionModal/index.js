import AccordionModal from './AccordionModal.vue'

export default AccordionModal
