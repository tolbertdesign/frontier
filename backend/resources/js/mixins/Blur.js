export default {
  methods: {
    blur () {
      document.getElementById('app').style.filter = 'blur(4px)'
    },
    unBlur () {
      document.getElementById('app').style.filter = 'none'
    },
  },
}
