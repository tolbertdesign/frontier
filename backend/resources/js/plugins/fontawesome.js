import Vue from 'vue'

import { library, dom } from '@fortawesome/fontawesome-svg-core'

// Brands (fab)
import { faFacebook, faFacebookF, faFacebookSquare, faTwitter } from '@fortawesome/free-brands-svg-icons'

// Regular (far)
import {
  faAddressCard,
  faArrowCircleRight,
  faBars as farBars,
  faCircle,
  faCommentAltLines,
  faEdit,
  faEnvelope,
  faExternalLinkSquare,
  faInfoCircle,
  faLock,
  faSignIn,
  faSignOut,
  faSquare,
  faTimes as farTimes,
  faUserEdit as farUserEdit,
} from '@fortawesome/pro-regular-svg-icons'

// Light (fal)
import { faBars, faTimes, faUserEdit } from '@fortawesome/pro-light-svg-icons'

// Solid Style Icons (fas)
import {
  faBars as fasBars,
  faBell,
  faBold,
  faCaretDown,
  faCaretUp,
  faCheck,
  faCheckSquare,
  faChevronDown,
  faChevronUp,
  faChevronLeft,
  faChevronRight,
  faCircle as fasCircle,
  faCode,
  faCoffee,
  faCog,
  faComment,
  faEdit as fasEdit,
  faEnvelope as fasEnvelope,
  faExclamationCircle,
  faExclamationTriangle,
  faExpandArrowsAlt,
  faEye,
  faItalic,
  faLink,
  faListOl,
  faListUl,
  faLock as fasLock,
  faMinus,
  faMobileAlt,
  faParagraph,
  faPlayCircle,
  faPlus,
  faQuestionCircle,
  faQuoteRight,
  faRedo,
  faShare,
  faShareSquare,
  faSortDown,
  faSpinner,
  faStrikethrough,
  faTimes as fasTimes,
  faTrashAlt,
  faUnderline,
  faUndo,
  faUser,
  faUsers,
  faUserPlus,
} from '@fortawesome/free-solid-svg-icons'

import {
  FontAwesomeIcon,
  FontAwesomeLayers,
  FontAwesomeLayersText,
} from '@fortawesome/vue-fontawesome'

library.add(
  faAddressCard,
  faArrowCircleRight,
  faBars,
  faBell,
  faBold,
  faCaretDown,
  faCaretUp,
  faCheck,
  faCheckSquare,
  faChevronDown,
  faChevronUp,
  faChevronLeft,
  faChevronRight,
  faCircle,
  faCode,
  faCoffee,
  faCog,
  faComment,
  faCommentAltLines,
  faEdit,
  fasEdit,
  faEnvelope,
  faExclamationCircle,
  faExclamationTriangle,
  faExpandArrowsAlt,
  faExternalLinkSquare,
  faEye,
  faFacebook,
  faFacebookF,
  faFacebookSquare,
  faInfoCircle,
  faItalic,
  faLink,
  faListOl,
  faListUl,
  faLock,
  faMinus,
  faMobileAlt,
  faParagraph,
  faPlayCircle,
  faPlus,
  faQuestionCircle,
  faQuoteRight,
  faRedo,
  farBars,
  farTimes,
  fasBars,
  fasCircle,
  fasEnvelope,
  faShare,
  faShareSquare,
  faSignIn,
  faSignOut,
  fasLock,
  faSortDown,
  faSpinner,
  faSquare,
  fasTimes,
  faUsers,
  faStrikethrough,
  faTimes,
  faTrashAlt,
  faTwitter,
  faUnderline,
  faUndo,
  faUser,
  faUserEdit,
  faUserPlus,
  farUserEdit,
)

dom.watch()

Vue.component('FontAwesomeIcon', FontAwesomeIcon)
Vue.component('FontAwesomeLayers', FontAwesomeLayers)
Vue.component('FontAwesomeLayersText', FontAwesomeLayersText)
