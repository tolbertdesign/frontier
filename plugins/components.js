import Vue from 'vue'

/*
 ** Import all components from `base/` and `icon/` directories
 ** Learn more on require.context() on https://webpack.js.org/guides/dependency-management/#requirecontext
 */
function importComponents(r) {
  r.keys().forEach(filename => {
    let Component = r(filename)

    Component = Component.default || Component
    Component.name =
      Component.name || filename.replace(/^.+\//, '').replace(/\.\w+$/, '')
    Vue.component(Component.name, Component)
  })
}

importComponents(require.context('@/components/core', true, /\.(vue|js)$/))
importComponents(require.context('@/components/icons', true, /\.(vue|js)$/))
importComponents(require.context('@/components/examples', true, /\.(vue|js)$/))
