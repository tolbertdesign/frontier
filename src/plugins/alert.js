import Vue from 'vue'

import { VAlert } from 'vuetensils'

Vue.component('VAlert', VAlert)

// VueEssentials
