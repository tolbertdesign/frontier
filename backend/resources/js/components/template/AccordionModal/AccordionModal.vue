<template>
  <div class="max-h-screen overflow-y-auto shadow-lg">
    <div class="modal-card rounded-xl bg-white shadow-lg">
      <header class="modal-card-head bg-white relative items-start">
        <slot name="header">
          <p class="modal-card-title flex items-center justify-between text-xl font-weight-normal">
            <span style="width: 24px; height: 48px" />
            <span class="text-2xl flex-1 text-center font-bold capitalize">{{ header }}</span>
            <a
              href="#"
              class="text-black self-start"
              @click.prevent="closeModal"
            >
              <span class="icon">
                <FontAwesomeIcon
                  :icon="['fas', 'times']"
                />
              </span>
            </a>
          </p>
        </slot>
      </header>
      <section class="modal-card-body p-0 overflow-auto">
        <slot>
          <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
            <p class="mb-4">
              {{ content }}
            </p>
          </div>
        </slot>
      </section>
      <div class="card-footer">
        <div class="card-footer-item flex-col">
          <slot name="footer">
            {{ footer }}
          </slot>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Blur from '@/mixins/Blur'

export default {
  name: 'AccordionModal',
  mixins: [Blur],
  props: {
    header: {
      type: String,
      default: '',
    },
    content: {
      type: String,
      default: '',
    },
    footer: {
      type: String,
      default: '',
    },
  },
  methods: {
    closeModal () {
      this.$emit('close')
      this.$parent.$emit('close')
      this.unBlur()
    },
  },
}
</script>
