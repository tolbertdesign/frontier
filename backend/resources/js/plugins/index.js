import 'nprogress/nprogress.css'
// import '@fortawesome/fontawesome-free/css/all.min.css'

import '@/plugins/buefy'
import '@/plugins/fontawesome'
import '@/plugins/gravatar'
import '@/plugins/tooltips'
import '@/plugins/vue-croppa'
import '@/plugins/vue-meta'
import '@/plugins/vuelidate'

// import '@/plugins/validation'
