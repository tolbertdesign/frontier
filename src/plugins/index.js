import '@/plugins/gravatar'
import '@/plugins/vue-particles'
import '@/plugins/portal-vue'
import '@/plugins/tooltips'
// import '@/plugins/validation'
