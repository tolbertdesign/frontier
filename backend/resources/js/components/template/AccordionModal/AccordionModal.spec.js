import { shallowMount, createLocalVue } from '@vue/test-utils'
import AccordionModal from './AccordionModal.vue'
import Buefy from 'buefy'
import Blur from '@/mixins/Blur'
import '@/plugins/fontawesome'

const localVue = createLocalVue()
localVue.use(Buefy, {
  defaultIconPack: 'fas',
})

describe('AccordionModal', () => {
  it('Snapshop test for AccordionModal', () => {
    const wrapper = shallowMount(AccordionModal, {
      localVue,
      mixins: {
        Blur,
      },
      slots: {
        header: '<div>header slot</div>',
        default: '<div>default slot</div>',
        footer: '<div>footer slot</div>',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
