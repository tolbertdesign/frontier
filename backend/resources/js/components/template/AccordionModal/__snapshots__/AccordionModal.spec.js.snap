// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionModal Snapshop test for AccordionModal 1`] = `
<div class="max-h-screen overflow-y-auto shadow-lg">
  <div class="modal-card rounded-xl bg-white shadow-lg">
    <header class="modal-card-head bg-white relative items-start">
      <div>header slot</div>
    </header>
    <section class="modal-card-body p-0 overflow-auto">
      <div>default slot</div>
    </section>
    <div class="card-footer">
      <div class="card-footer-item flex-col">
        <div>footer slot</div>
      </div>
    </div>
  </div>
</div>
`;
